test/api/jd_by_pulse_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/block-explorer/blob/master/LICENSE.md.
+
+// +build heavy_mock_integration
+
+package api
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"testing"
+
+	"github.com/antihax/optional"
+	"github.com/insolar/block-explorer/instrumentation/converter"
+	"github.com/insolar/block-explorer/test/heavymock"
+	"github.com/insolar/block-explorer/test/integration"
+	"github.com/insolar/block-explorer/testutils"
+	"github.com/insolar/insolar/ledger/heavy/exporter"
+	"github.com/insolar/insolar/pulse"
+	"github.com/insolar/spec-insolar-block-explorer-api/v1/client"
+	"github.com/stretchr/testify/require"
+)
+
+const defaultLimit = 20
+
+func TestGetJetDropsByPulse(t *testing.T) {
+	ts := integration.NewBlockExplorerTestSetup(t).WithHTTPServer(t)
+	defer ts.Stop(t)
+
+	pulsesCount := 5
+	jetDropsCount := 2
+	records := testutils.GenerateRecordsWithDifferencePulsesSilence(pulsesCount, jetDropsCount)
+	pulses := make(map[pulse.Number]bool, pulsesCount)
+	jds := make(map[string]bool, pulsesCount*2)
+	for _, r := range records {
+		pulses[r.Record.ID.GetPulseNumber()] = false
+		jds[converter.JetIDToString(r.Record.JetID)] = false
+	}
+
+	require.NoError(t, heavymock.ImportRecords(ts.ConMngr.ImporterClient, records))
+	lastPulse := records[len(records)-1].Record.ID.GetPulseNumber()
+	lastRecordInPulse := []*exporter.Record{testutils.GenerateRecordInNextPulse(lastPulse)}
+	require.NoError(t, heavymock.ImportRecords(ts.ConMngr.ImporterClient, lastRecordInPulse))
+
+	ts.WaitRecordsCount(t, len(records), 5000)
+
+	c := GetHTTPClient()
+	pulsesResp, err := c.Pulses(t, nil)
+	require.NoError(t, err)
+	require.Len(t, pulsesResp.Result, pulsesCount)
+	var i int
+
+	t.Run("check received data in jetdrops", func(t *testing.T) {
+		t.Log("C5223 Get Jet drops by Pulse number")
+		for p := range pulses {
+			response, err := c.JetDropsByPulseNumber(t, int64(p), nil)
+			require.NoError(t, err)
+			require.Equal(t, int64(jetDropsCount), response.Total)
+			require.Len(t, response.Result, jetDropsCount)
+			require.Empty(t, response.Message)
+			require.Empty(t, response.ValidationFailures)
+			for _, jd := range response.Result {
+				require.Equal(t, int64(p), jd.PulseNumber)
+				require.Equal(t, int64(1), jd.RecordAmount)
```

Comment:
I think the record number should be 2

Reply:
Please explain what do you mean? precondition creates pulses -> each pulse contains two JetDrops -> each JetDrops contains one record. This line checks if RecordAmount in JetDrops is (1) as expected